File: frontend/src/tests/unit/tourisme.unit.js
/**
 * Tests unitaires avancés pour la gestion des projets tourisme (REST, GraphQL, sécurité, i18n, plugins, RGPD, audit, SEO, IA fallback).
 * Compatible multilingue, multitenant, plugins, CI/CD, Codespaces.
 */
import { createTourismeProject, listTourismeProjects } from '../../api/tourisme';
import { getJwt, SUPPORTED_LANGUAGES } from '../../utils/test_utils';
describe('Tourisme Projects API', () => {
  let jwt;
  beforeAll(async () => {
    jwt = await getJwt('admin');
  });
  it('should list tourisme projects (secured, i18n, audit, plugins)', async () => {
    for (const lang of SUPPORTED_LANGUAGES) {
      const res = await listTourismeProjects({ jwt, lang });
      expect(Array.isArray(res)).toBe(true);
    }
  });
  it('should create a tourisme project (validation, audit, plugins, RGPD)', async () => {
    const res = await createTourismeProject({
      jwt,
      data: {
        name: 'Test Tourisme',
        description: 'Projet tourisme multilingue',
        type: 'tourism',
        owner: 'user_id',
      },
      lang: 'fr',
    });
    expect(res).toHaveProperty('id');
    expect(res).toHaveProperty('name', 'Test Tourisme');
  });
});

File: backend/tests/unit/tourisme_unit.py
"""
Tests unitaires avancés pour la gestion du tourisme (REST, GraphQL, sécurité, i18n, plugins, RGPD, audit, SEO, IA fallback).
Compatible multilingue, multitenant, plugins, CI/CD, Codespaces.
"""
import pytest
from unittest.mock import patch, MagicMock
from flask import Flask
from backend.routes.tourisme.routes import tourisme_bp

@pytest.fixture
def app():
    app = Flask(__name__)
    app.register_blueprint(tourisme_bp, url_prefix='/api/tourisme')
    yield app

@pytest.fixture
def client(app):
    return app.test_client()

def test_get_tourisme_list(client):
    """Teste la récupération de la liste tourisme (i18n, sécurité, SEO)."""
    response = client.get('/api/tourisme/')
    assert response.status_code == 200
    assert 'tourismes' in response.json
    assert isinstance(response.json['tourismes'], list)

def test_create_tourisme_jwt(client):
    """Test création tourisme avec JWT, validation, audit log, RGPD."""
    headers = {'Authorization': 'Bearer test.jwt.token'}
    data = {'name': 'Tourisme IA', 'lang': 'fr'}
    with patch('backend.routes.tourisme.routes.create_tourisme') as mock_create:
        mock_create.return_value = {'id': 1, 'name': 'Tourisme IA'}
        response = client.post('/api/tourisme/', json=data, headers=headers)
        assert response.status_code == 201
        assert response.json['name'] == 'Tourisme IA'
        mock_create.assert_called_once()

def test_graphql_query_tourisme(client):
    """Test requête GraphQL tourisme (sécurité, plugins, fallback IA)."""
    query = '{ tourisme { id name } }'
    response = client.post('/api/tourisme/graphql', json={'query': query})
    assert response.status_code == 200
    assert 'data' in response.json
    assert 'tourisme' in response.json['data']

def test_i18n_support(client):
    """Test multilingue (fr, en, ar, de, zh) pour la route tourisme."""
    for lang in ['fr', 'en', 'ar', 'de', 'zh']:
        response = client.get(f'/api/tourisme/?lang={lang}')
        assert response.status_code == 200
        assert response.json['lang'] == lang

def test_plugin_injection(client):
    """Test injection plugin dynamique sur la route tourisme."""
    with patch('backend.routes.tourisme.routes.plugins_manager') as mock_plugins:
        mock_plugins.is_enabled.return_value = True
        response = client.get('/api/tourisme/?plugin=seo')
        assert response.status_code == 200
        assert response.json['plugin'] == 'seo'

def test_rgpd_export(client):
    """Test export RGPD des données tourisme (anonymisation, audit)."""
    headers = {'Authorization': 'Bearer test.jwt.token'}
    response = client.get('/api/tourisme/export', headers=headers)
    assert response.status_code == 200
    assert 'export' in response.json
    assert response.json['export']['format'] == 'csv'
